includes/chart.js

```diff
@@ -0,0 +1,330 @@
+// chart.js  - ANNO DOMINI 2024 - eScience Center, AMS
+// Authors - Ermanno Lo Cascio, Suvayu Ali, Sander van Rijn, Ole Mussmann
+
+/*
+                              o
+          ######    #######  ###   ######    ####     ##   #######  #########
+######    ##       ##        ###   ##        ## ##    ##  ##        ##
+#    ##   ##       ##        ###   ##        ##  ##   ##  ##        ##
+######    #######  ##        ###   ######    ##   ##  ##  ##        #########
+#              ##  ##        ###   ##        ##    #####  ##        ##
+######    #######  ########  ###   #######   ##     ####  ########  #########   Center Amsterdam
+-----------------------------------------------------------------------------
+
+// NOTES
+
+The chart.js contains all the functions to render the svg flowchart based on data uploaded.
+The operations executed here are:
+    1. Generate the nodes and links arrays
+    2. Create the nodes and set all the required attributes
+    3. Generata the links
+
+*/
+
+// Define nodes following the template
+const row_template = {
+    nodes: [
+        {
+            id: 1, x: 0, y: 30, color: '#ffc70f', shape: 'polygon', type: 'activationCondition',
+            points: '30,60 9,30 30,0 90,0 111,30 90,60',
+        },
+        {
+            id: 2, x: 120, y: 30, color: '#0fafff', shape: 'circle', type: 'attribute',
+            size: '5px', radiusX: 48, radiusY: 36,
+        },
+        {
+            id: 3, x: 240, y: 0, color: '#ffc70f', shape: 'rect', type: 'aim',
+            width: 100, height: 60,
+        },
+        {
+            id: 4, x: 420, y: 0, color: '#0fafff', shape: 'rect', type: 'directObject',
+            width: 110, height: 60, rx: 30, ry: 30,  // rx/ry for rounded corners
+        },
+        {
+            id: 5, x: 240, y: 90, color: '#0fafff', shape: 'rect', type: 'executionConstraint',
+            width: 110, height: 60, rx: 30, ry: 30,  // rx/ry for rounded corners
+        },
+        {
+            id: 6, x: 420, y: 90, color: '#0fafff', shape: 'rect', type: 'indirectObject',
+            width: 110, height: 60, rx: 30, ry: 30,  // rx/ry for rounded corners
+        },
+    ],
+    links: [
+        { source: 1, target: 2 },
+        { source: 2, target: 3 },  // include deontic info here
+        { source: 3, target: 4 },
+        { source: 5, target: 3 },
+        { source: 4, target: 6 },
+    ],
+};
+
+const colors = { formal: '#99cc00', informal: '#f4b183' }
+
+const rowAttributes = [
+    "id",
+    "statementType",
+    "attribute",
+    "deontic",
+    "aim",
+    "directObject",
+    "typeOfDirectObject",
+    "indirectObject",
+    "typeOfIndirectObject",
+    "activationCondition",
+    "executionConstraint",
+    "orElse"
+];
+
+function addNodesAndLinks(rowValues) {
+    var svgContainer = document.getElementById("svgContainer");
+    var svg = svgContainer.querySelector("svg");
```

Comment:
Most of the `var` declarations in this PR could probably be `let`'s. I don't know enough about web dev, but I would imagine it to be safer to limit the scope to where it's needed. I'll skip the following ones, so please check where it's applicable.

https://stackoverflow.com/questions/762011/what-is-the-difference-between-let-and-var